opensearch_py_ml/ml_models/metrics_correlation/nmf.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+from typing import Tuple
+
+import torch
+
+
+class NMF(torch.nn.Module):
```

Comment:
We previously discussed whether this should be a `nn.Module` on its own vs. just a function called during `MCorr.forward()`. What's the current thinking on that?

Reply:
Yeah, for now I'm keeping this way.